cpp/src/aztec3/circuits/rollup/base/.test.cpp

```diff
@@ -242,24 +266,145 @@ TEST_F(base_rollup_tests, contract_leaf_inserted)
     };
     inputs.kernel_data[0].public_inputs.end.new_contracts[0] = new_contract;
 
+    auto empty_contract_tree = native_base_rollup::MerkleTree(CONTRACT_TREE_HEIGHT);
+    // fetch sibling path from hash path (only get the second half of the hash path)
+    auto hash_path_of_0 = empty_contract_tree.get_hash_path(0);
+    std::array<NT::fr, CONTRACT_SUBTREE_INCLUSION_CHECK_DEPTH> sibling_path_of_0;
+    for (size_t i = 0; i < CONTRACT_SUBTREE_INCLUSION_CHECK_DEPTH; ++i) {
+        sibling_path_of_0[i] = hash_path_of_0[CONTRACT_SUBTREE_DEPTH + i].second;
+    }
+    // Set the new_contracts_subtree_sibling_path
+    inputs.new_contracts_subtree_sibling_path = sibling_path_of_0;
+
+    // manually create expected end contract tree snapshot
+    auto expected_contract_leaf = crypto::pedersen_hash::hash_multiple(
+        { new_contract.contract_address, new_contract.portal_contract_address, new_contract.function_tree_root });
+    auto expected_contracts_subtree = stdlib::merkle_tree::MemoryTree(CONTRACT_SUBTREE_DEPTH);
+    expected_contracts_subtree.update_element(0, expected_contract_leaf);
+    // manually iterate through sibling path to create end snapshot
```

Comment:
Why are you manually iterating through it instead of inserting into a memory tree?